src/cache.rs

```diff
@@ -0,0 +1,90 @@
+//! Prepared statements cache for faster execution.
+extern crate lru_cache;
+
+use {Result, Connection, Statement};
+use self::lru_cache::LruCache;
+
+/// Prepared statements cache.
+///
+/// FIXME limitation: the same SQL can be cached only once...
+#[derive(Debug)]
+pub struct StatementCache<'conn> {
+    pub conn: &'conn Connection,
+    cache: LruCache<String, Statement<'conn>>,
+}
+
+impl<'conn> StatementCache<'conn> {
+    /// Create a statement cache.
+    pub fn new(conn: &'conn Connection, capacity: usize) -> StatementCache<'conn> {
+        StatementCache {
+            conn: conn,
+            cache: LruCache::new(capacity),
+        }
+    }
+
+    /// Search the cache for a prepared-statement object that implements `sql`.
+    // If no such prepared-statement can be found, allocate and prepare a new one.
+    ///
+    /// # Failure
+    ///
+    /// Will return `Err` if no cached statement can be found and the underlying SQLite prepare call fails.
+    pub fn get(&mut self, sql: &str) -> Result<Statement<'conn>> {
```

Comment:
I'm inclined to think this should return a `Result<CachedStatement<'cache>>`. `CachedStatement` would wrap up `Statement` (and implement `Deref`/`DerefMut` for it too, I think), and would return it to the cache in `drop`. This would remove the need to remember to call `release`. We could actually remove `release` altogether, I think, and add a `release` or `discard` method to `CachedStatement` that would (a) take `self` (to consume it) and (b) set a flag (or something similar) to avoid returning the statement to the cache in `drop`.

What do you think?